File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    @room = Room.find(params[:room_id])
    stream_for @room # Παρακολουθούμε τα μηνύματα του room
  end

  def unsubscribed
    # Καθαρισμός όταν ο χρήστης αποσυνδεθεί
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer";

const roomId = document.getElementById("messages").dataset.roomId;

const ChatChannel = consumer.subscriptions.create(
  { channel: "ChatChannel", room_id: roomId },
  {
    // Όταν φτάνει το μήνυμα
    received(data) {
      const messagesContainer = document.getElementById("messages");
      if (messagesContainer) {
        messagesContainer.innerHTML += data.message; // Προσθέτει το νέο μήνυμα
      }
    },

    // Συνδρομή σε συγκεκριμένο κανάλι για το room_id
    connected() {
      console.log('Connected to the ChatChannel');
    },

    disconnected() {
      console.log('Disconnected from the ChatChannel');
    }
  }
);

export default ChatChannel;
